tests/breed-hpprocurveN1178Test.php

```diff
@@ -0,0 +1,143 @@
+
+class hpprocurveN1178Test extends RTTestCase
+{
+
+	// hpprocurveN1178Read8021QConfig
+	/**
+	 * @dataProvider providerHP8021QConfigSamples
+	 */
+
+	public function testHP8021QConfParser ($test_data)
+	{
+	foreach ($test_data as $test => $test_array)
+		{
+		$input = $test_array['input'];
+		$expected = $test_array['results'];
+		$test_descr = $test ." " . $test_array['descr'];
+		$actual = hpprocurveN1178Read8021QConfig ($input);
+		if (! $this->assertEquals($expected, $actual))
+			print_r("\n\tOK! ----> HP 802.1Q Config Parser test " . $test_descr);
+		}
+	}
+
+
+	// hpprocurveN1178ReadInterfaceStatus
+	/**
+	 * @dataProvider providerHPShowIntBriefSamples
+	 */
+
+	public function testHPIntBriefConfParser ($test_data)
+	{
+	foreach ($test_data as $test => $test_array)
+		{
+		$input = $test_array['input'];
+		$expected = $test_array['results'];
+		$test_descr = $test ." " . $test_array['descr'];
+		$actual = hpprocurveN1178ReadInterfaceStatus ($input);
+		if (! $this->assertEquals($expected, $actual))
+			print_r("\n\tOK! ----> HP Show Interface Brief Parser test " . $test_descr);
+		}
+	}
+
+
+	// hpprocurveN1178ReadLLDPStatus
+	/**
+	 * @dataProvider providerHPLLDPSamples
+	 */
+
+	public function testHPLLDP ($test_data)
+	{
+	foreach ($test_data as $test => $test_array)
+		{
+		$input = $test_array['input'];
+		$expected = $test_array['results'];
+		$test_descr = $test ." " . $test_array['descr'];
+		$actual = hpprocurveN1178ReadLLDPStatus ($input);
+		if (! $this->assertEquals($expected, $actual))
+			print_r("\n\tOK! ----> HP LLDP Parser test " . $test_descr);
+		}
+	}
+
+	// hpprocurveN1178ReadMacList
+	/**
+	 * @dataProvider providerHPMACAddrSamples
+	 */
+
+	public function testHPReadMacList ($test_data)
+	{
+	foreach ($test_data as $test => $test_array)
+		{
+		$input = $test_array['input'];
+		$expected = $test_array['results'];
+		$test_descr = $test ." " . $test_array['descr'];
+		$actual = hpprocurveN1178ReadMacList ($input);
+		if (! $this->assertEquals($expected, $actual))
+			print_r("\n\tOK! ----> HP Read MAC List Parser test " . $test_descr);
+		}
+	}
+
+
+	// hpprocurveN1178ReadLLDPStatus
+	/**
+	 * @dataProvider providerHP8021QSamples
+	 */
+ 
+	public function testHPvlanMGMT ($test_data)
+	{
+	foreach ($test_data as $test => $test_array)
+		{
+		$dummy_object_id = $test_array['input']['dummy_object_id'];
+		$queue = $test_array['input']['queue'];
+		$dummy_vlan_names = $test_array['input']['dummy_vlan_names'];
+		$expected = $test_array['results'];
+		$test_descr = $test ." " . $test_array['descr'];
+		$actual = hpprocurveN1178TranslatePushQueue ($dummy_object_id, $queue, $dummy_vlan_names);
+		if (! $this->assertEquals($expected, $actual))
+			print_r("\n\tOK! ----> HP 8021Q test " . $test_descr);
+		}
+	}
+
+	public function providerHP8021QConfigSamples ()
+	{
+	$data_file = 'breeds_data/breed-hpprocurveN1178-8021QConfigSamples.php';
+	if(is_file($data_file)) 
+		return include $data_file;
```

Comment:
The array with the samples should be in the provider function, unless there is a good reason to do otherwise.

Reply:
Also please put the new files in `tests`, this way `express.sh` will test them for syntax errors together with the new files.

Reply:
The first version was done as you commented above, but the file was huge and it was quite difficult to debug it (even when the array had one-line format). If you still recommends to place sample data into file breed-hpprocurveN1178Test.php, i will update it.

Reply:
Same as above please (new directories have maintenance cost).